cpp/include/raft/random/detail/make_regression.cuh

```diff
@@ -40,7 +40,10 @@ namespace detail {
 
 /* Internal auxiliary function to help build the singular profile */
 template <typename DataT, typename IdxT>
-static __global__ void _singular_profile_kernel(DataT* out, IdxT n, DataT tail_strength, IdxT rank)
+static _RAFT_KERNEL void _singular_profile_kernel(DataT* out,
```

Comment:
Since we have `#define _RAFT_KERNEL static __global__` - doesn't this expand out to `static static __global__` ? should we drop the `static` here (and in other similar places)?

```suggestion 
_RAFT_KERNEL void _singular_profile_kernel(DataT* out,
```

Reply:
@benfred the whole point of this macro is that according to #1722, we need to define _all_ `__global__` to be `static __global__`. I'd prefer not to have to specify `static` explicitly when it's going to have to be specified everywhere anyways.

Reply:
EDIT: Don't mind me, I misread your comment. Yeah absolutely what you found here was a classic find and replace error.

Reply:
I think you misunderstood me - I'm on board with changing `__global__` to `_RAFT_KERNEL`.  

I'm just saying the `static _RAFT_KERNEL` should probably be just `_RAFT_KERNEL` (so that we're not specifying static twice)

Reply:
> I think you misunderstood me - I'm on board with changing __global__ to _RAFT_KERNEL.

I'm just saying the static _RAFT_KERNEL should probably be just _RAFT_KERNEL (so that we're not specifying static twice)

Yes, absolutely I misunderstood and on second inspection realized that. Yes you are right, I did a search and replace for `__global__` but I didn't actually check for places that already had `static __global__`.